src/AMAMM.sol

```diff
@@ -2,69 +2,208 @@
-import {SafeCast} from "v4-core/libraries/SafeCast.sol";
+import {IHooks} from "v4-core/interfaces/IHooks.sol";
 
-contract AMAMM is BaseHook {
-    using SafeCast for uint256;
+contract AMAMM is IAmAmm {
+    constructor() public {}
 
     // TODO: set this value to the ePoch swap fee
     uint128 public constant SWAP_FEE_BIPS = 123; // 123/10000 = 1.23%
     uint128 public constant TOTAL_BIPS = 10000;
 
-    constructor(IPoolManager poolManager) BaseHook(poolManager) {}
-
-    function getHookPermissions()
-        public
-        pure
-        override
-        returns (Hooks.Permissions memory)
-    {
-        return
-            Hooks.Permissions({
-                beforeInitialize: false,
-                afterInitialize: false,
-                beforeAddLiquidity: false,
-                afterAddLiquidity: false,
-                beforeRemoveLiquidity: false,
-                afterRemoveLiquidity: false,
-                beforeSwap: false,
-                afterSwap: true, // Override how swaps are done
-                beforeDonate: false,
-                afterDonate: false,
-                beforeSwapReturnDelta: false,
-                afterSwapReturnDelta: true, // Allow afterSwap to return a custom delta
-                afterAddLiquidityReturnDelta: false,
-                afterRemoveLiquidityReturnDelta: false
-            });
+    /// -----------------------------------------------------------------------
+    /// Library usage
+    /// -----------------------------------------------------------------------
+
+    using SafeCastLib for *;
+    using FixedPointMathLib for *;
+
+    /// -----------------------------------------------------------------------
+    /// Constants
+    /// -----------------------------------------------------------------------
+
+    function K(PoolId) public view virtual returns (uint40) {
+        return 24;
+    }
+
+    function EPOCH_SIZE(PoolId) public view virtual returns (uint256) {
+        return 1 hours;
+    }
+
+    function MIN_BID_MULTIPLIER(PoolId) public view virtual returns (uint256) {
+        return 1.1e18;
+    }
+
+    /// -----------------------------------------------------------------------
+    /// Storage variables
+    /// -----------------------------------------------------------------------
+
+    mapping(PoolId id => uint40) internal _lastUpdatedEpoch;
+    mapping(address deposits => uint256) public _userBalance;
+    mapping(Currency currency => uint256) internal _totalFees;
+    mapping(PoolId id => mapping(uint40 => Bid)) public poolEpochManager;
+    mapping(address manager => mapping(Currency currency => uint256)) internal _fees;
+    mapping(PoolId id => address) internal _bidToken;
+
+    /// -----------------------------------------------------------------------
+    /// Getter actions
+    /// -----------------------------------------------------------------------
+
+    function getCurrentManager(PoolId id, uint40 epoch) public view returns (Bid memory) {
```

Comment:
```suggestion
```
I think should always use `getLastManager` as the `getCurrentManager` could return wrong result that shouldn't be used at all.

Reply:
Enhancement: I thought it was removed.